include/util/geojson_debug_logger.hpp

```diff
@@ -0,0 +1,185 @@
+            if (!first)
+                ofs << ",\n\t\t";
+
+            util::json::render(ofs, object.get<util::json::Object>());
+
+            first = false;
+        }
+    }
+
+    // writes a single feature into the Geojson file
+    template <typename... Args> static bool Write(Args &&... args)
+    {
+        // make sure to syncronize logging output, our writing should be sequential
+        std::lock_guard<std::mutex> guard(lock);
+
+        // if there is no logfile, we cannot write
+        if (!ofs.is_open() || (nullptr == policy))
+        {
+            // this can only happend between two guards when concurrent writing occurs
+            return false;
+        }
+
+        // use our policy to convert the arguments into geojson, this can be done in parallel
+        const auto json_object = (*policy)(std::forward<Args>(args)...);
+
+        // different features are separated by `,`
+        // since we are not building a full json collection of features, we have to do some of the
+        // bookeeping ourselves. This prevens us from doing one huge json object, if we are
+        // processing larger results
+        output(first, json_object);
+        first = false;
+
+        return static_cast<bool>(ofs);
+    }
+
+    // Opening a logger, we initialize a geojson feature collection
+    // to be later filled with additional data
+    static bool Open(const std::string &logfile)
+    {
+        // if a file is open, close it off. When this function returns, there should be an open
+        // logfile. However, there is a brief period between closing and openen where we could miss
+        // out on log output. Such a sad life
+        if (ofs.is_open())
+        {
+            util::SimpleLogger().Write(logWARNING)
+                << "Overwriting " << logfile
+                << ". Is this desired behaviour? If this message occurs more than once rethink the "
+                   "location of your Logger Guard.";
+            Close();
+        }
+
+        // make sure to syncronize logging output, cannot be locked earlier, since Close also locks
+        // and we don't want deadlocks
+        std::lock_guard<std::mutex> guard(lock);
+        ofs.open(logfile, std::ios::binary);
+
+        // set up a feature collection
+        ofs << "{\n\t\"type\": \"FeatureCollection\",\n\t\"features\": [\n\t";
+        // remember whether we need to output a colon
+        first = true;
+
+        return static_cast<bool>(ofs);
+    }
+
+    // finalising touches on the GeoJson
+    static bool Close()
+    {
+        // make sure to syncronize logging output
+        std::lock_guard<std::mutex> guard(lock);
+
+        // finishe the geojson feature collection and close it all off
+        if (ofs.is_open())
+        {
+            ofs << "\n\t]\n}";
+            ofs.close();
+        }
+
+        return static_cast<bool>(ofs);
+    }
+
+    static void SetPolicy(geojson_conversion_policy *new_policy) { policy = new_policy; }
```

Comment:
Should this always overwrite an already existing policy? Or should we 
- assert !policy
- assert new_poliy